app/models/filters/filter_base.rb

```diff
@@ -156,6 +156,8 @@ def update(filters) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/Perc
       self.cohort_column_matched_date = filters.dig(:cohort_column_matched_date)&.presence || cohort_column_matched_date
 
       self.disabilities = filters.dig(:disabilities)&.reject(&:blank?)&.map(&:to_i).presence || disabilities
+      # Exclude HIV/AIDS from disabilities unless the user can view HIV/AIDS status
+      self.disabilities = disabilities.reject { |id| id == 8 && !user.can_view_hiv_status? } if disabilities.include?(8)
```

Comment:
what do you think about something like this to DRY up the logic?
```suggestion
      self.disabilities = disabilities.select { |id| available_disabilities.values.include?(id) } if disabilities.present?
```

Reply:
Excellent.  Thanks